File: SkillTracker/EngineerService/Models/SkillProfile.cs
using EngineerService.Validators;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace EngineerService.Models
{
    public class SkillProfile
    {
        [BsonId]
        public ObjectId Id { get; set; }

        [Required]
        public Associate AssociateInfo { get; set; }

        [Required]
        [SkillList]
        public IList<Skill> SkillInfo { get; set; }

        public DateTime AddedDate { get; set; }

        public DateTime LastUpdated { get; set; }

        public SkillProfile()
        {
            this.AddedDate = DateTime.Now;
        }
    }
}

File: skilltracker-app/src/models/skill-profile.ts
import { AssociateDetails } from "./associate-details";
import { Skill } from "./skill";

export class SkillProfile {
    public id!: number;
    public associateInfo!: AssociateDetails;
    public skillInfo!: Array<Skill>;
    public addedDate!: Date;
    public lastUpdatedDate!: Date;
}